### PR TITLE
Delete Datastore Table Button

### PR DESCRIPTION
**Pull request by JVickery-TBS**
feat(views): added view and button to delete ds table;

    Added view to delete datastore table.
    Added button to delete a resource's datastore table.

Adds a button to the Xloader data page to delete a Resource's datastore table.

### CODE REVIEW

File: ckanext/xloader/templates/xloader/resource_data.html [hunk 1]
```diff
@@ -5,10 +5,24 @@
...
+  {% block delete_ds_button %}
+    <form method="post" action="{{ delete_action }}" class="mb-3 d-inline-block">
+      {{ h.csrf_input() if 'csrf_input' in h }}
+      <a href="{{ h.url_for('xloader.delete_datastore_table', id=pkg.id, resource_id=res.id) }}"
+        class="btn btn-danger pull-left"
+        type="submit"
+        data-module="confirm-action"
+        data-module-with-data=true
+        data-module-content="{{ _('Are you sure you want to delete this DataStore table and Data Dictionary?') }}"
+        >{% block delete_datastore_button_text %}<i class="fa fa-remove"></i>{{ _('Delete DataStore table') }}{% endblock %}</a>
```
Comment (ThrawnCA) [inline ckanext/xloader/templates/xloader/resource_data.html:20]: I think including the word "table" here may be exposing more of the internal implementation than we want. The word isn't used elsewhere in the Datastore UI.

**Review by ThrawnCA — CHANGES_REQUESTED** [review]

**Comment by ThrawnCA** [comment]
This looks useful to have, particularly in combination with the heuristics we've added around choosing between Tabulator and fast copy.

File: ckanext/xloader/templates/xloader/resource_data.html (hunk 1 above)
Comment (JVickery-TBS) [reply ckanext/xloader/templates/xloader/resource_data.html]: @ThrawnCA this makes sense. I have removed any "table" from user visible things here, including the URL for the new view method

**Comment by JVickery-TBS** [comment]
@ThrawnCA for this PR, I also reverted the `ckanapi` dependency addition and used `get_action` from the toolkit instead

File: ckanext/xloader/views.py [hunk 2]
```diff
@@ -21,3 +23,36 @@ def resource_data(id, resource_id):
         except ValueError:
             rows = None
     return utils.resource_data(id, resource_id, rows)
+
+
+@xloader.route("/dataset/<id>/delete-datastore/<resource_id>", methods=("GET", "POST"))
+def delete_datastore_table(id, resource_id):
```
Comment (ThrawnCA) [inline ckanext/xloader/views.py:29]: Why does this need the package ID?

Comment (duttonw) [reply ckanext/xloader/views.py:29]: its to continue to match the comment pattern on all other queries.

Comment (ThrawnCA) [reply ckanext/xloader/views.py:29]: Fair enough. I would suggest, then, that there should be a sanity check to ensure that the resource actually belongs to the specified package.

Comment (JVickery-TBS) [reply ckanext/xloader/views.py:29]: Okay can add that check, the package id is also needed for the redirect to `xloader.resource_data`

**Review by duttonw — APPROVED** [review]

File: ckanext/xloader/views.py (hunk 2 above)
Comment (ThrawnCA) [reply ckanext/xloader/views.py:29]: Ah right, missed that one.

Comment (JVickery-TBS) [reply ckanext/xloader/views.py:29]: @ThrawnCA okay, now asserting the resource ownership

**Review by duttonw — APPROVED** [review]
👍

**Review by ThrawnCA — APPROVED** [review]

**Pull request merged**